### DISCUSSION TITLE
Does Directus work with image CDN providers?

### DISCUSSION THREAD

**Discussion by notflip** [directus/directus#17592] [Archive]
I'm wondering if Directus can work together with image CDN providers such as Imagekit, so that image compression, cropping etc can be applied to images uploaded in the CMS?

**Comment by br41nslug** [comment]
Directus does provide editing options like cropping and scaling out of the box 😄 https://docs.directus.io/app/file-library.html#edit-an-image

**Reply by notflip** [reply]
Right thanks! How about image compression? I think that's the most important one. What if a client uploads a 4MB image?

**Reply by domdus** [reply]
Directus offers image transformations on file access, produced out of the uploaded original file.
You can configure dimensions, formats, cropping and quality (compression) for the result image.
https://docs.directus.io/reference/files.html
https://docs.directus.io/configuration/project-settings.html#files-thumbnails

The final url (e.g. example.com/assets/1ac73658-8b62-4dea-b6da-529fbc9d01a4?fit=cover&width=200&height=200&quality=80 or example.com/assets/file-id?key=preset-key) is the thing you would want to "just cache" on a CDN.
You could run ImageKit transformations on top of that, but it may interferes.

**Reply by notflip** [reply]
Thanks for the information @domdus, I suppose the browser will cache these images by default, so there's no need for an external image caching service?
